docs/pages/reference/terraform/index.mdx

```diff
@@ -0,0 +1,67 @@
+---
+title: "Teleport Terraform Provider"
+description: |-
+  
+---
+
+The Teleport Terraform provider allows Terraform users to configure Teleport
+from Terraform.
+
+This section is the Teleport Terraform Provider reference.
+It lists all the supported resources and their fields.
+
+<Admonition type="tip">
+To get started with the Terraform provider, you must start with the installation guide.
+Once you got a working provider, we recommend you to follow the
+["Managing users and roles with IaC"](../management/dynamic-resources/user-and-role.mdx) guide.
+</Admonition>
+
+The provider exposes Teleport resources both as Terraform datasources and Terraform resources.
+Datasources are used by Terraform to fetch information from Teleport, while resources are used
+to create resources in Teleport.
+
+- [list of supported datasources](./datasources/index.mdx)
+- [list of supported resources](./resources/index.mdx)
+
+## Example Usage
+
+```hcl
+terraform {
+  required_providers {
+    teleport = {
+      source  = "terraform.releases.teleport.dev/gravitational/teleport"
+      version = "~> 15.0"
+    }
+  }
+}
+
+provider "teleport" {
+  # Update addr to point to Teleport Auth/Proxy
+  # addr              = "auth.example.com:3025"
+  addr               = "proxy.example.com:443"
+  identity_file_path = "terraform-identity/identity"
+}
+```
+
+{/*  schema generated by tfplugindocs */}
+## Schema
+
+### Optional
+
```

Comment:
I would add more context to this part of the intro page to help a user get started with configuring this block. At the moment, there's a usage example and a list of optional fields, and it's not clear when a user would want to use each of these optional fields.

Reply:
Done